app/models/estimate.rb

```diff
@@ -18,8 +19,22 @@ def total_cost
   private
 
   def expires_at_cannot_be_in_the_past
-    if expires_at.blank? || expires_at < Time.zone.now
+    return if expires_at.blank?
+
+    if expires_at < Time.zone.now
       errors.add(:expires_at, I18n.t("activerecord.errors.models.estimate.attributes.expires_at.in_the_past"))
     end
   end
+
+  def validate_estimate_details
+    estimate_details.each do |estimate_detail|
+      next if estimate_detail.marked_for_destruction?
+      next if estimate_detail.valid?
+
+      errors.delete(:"estimate_details.base")
+      estimate_detail.errors.each do |error|
+        errors.add(:base, "#{estimate_detail.product.name}: #{error.message}")
```

Comment:
aren't these errors shown in the estimate details table? I think it's enough to have them there